openmdao/recorders/case.py

```diff
@@ -297,6 +297,44 @@ def __init__(self, filename, counter, iteration_coordinate, timestamp, success,
         self.rel_err = rel_err
 
 
+class ProblemCase(Case):
```

Comment:
I may be missing something, but it seems like we have a bunch of Case wrappers in this file that don't really do anything but reorder some args to __init__ or maybe set some to None.  Just curious why the arg ordering is different between Case and some of the wrappers, and why we couldn't give some of the Case.__init__ args default values of None to eliminate the need for some of this code.

Reply:
I think this is a problem that evolved over the course of a couple PRs. I think `ProblemCase` is the only one that has to put `None` into its `super` call at the moment. But indeed, with re-ordering and a default value for Inputs we could reduce the code/comments for every case type (except SolverCase which stores a couple additional values).

Reply:
I put in a chore story to do this